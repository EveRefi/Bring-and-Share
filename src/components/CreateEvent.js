import React, { useState } from "react";
import axios from "axios";
import "../styles/CreateEvent.css";
import Alert from "./Alert";

const CreateEvent = () => {
  const initialState = {
    fields: {
      eventName: "",
      hostName: "",
      date: "",
      time: "",
      address: "",
    },
    alert: {
      message: "",
      isSuccess: false,
    },
  };

  const [fields, setFields] = useState(initialState.fields);
  const [alert, setAlert] = useState(initialState.alert);




  const handleCreateEvent = (e) => {
    e.preventDefault();
    setAlert({ message: "", isSuccess: false });
    
    const successAlertMessage = (url) => (
      <div>Click <a href={url}>here</a> to see your event!</div>
    )
    axios
      .post("https://final-mcrcodes-project.herokuapp.com/events", fields)
      .then((response) => {
       
        setAlert({
          message: successAlertMessage(`/events/${response.data.id}`),
          isSuccess: true,
        });
      })
      .catch((error) => {
        console.log(error);
        setAlert({
          message:
            "Something went wrong! Your event was not created. Please try again.",
          isSuccess: true,
        });
      });
  };

  const handleFieldChange = (e) => {
    setFields({ ...fields, [e.target.name]: e.target.value });
  };

  return (
    <div className="createEvent">
      <h2>Create your event</h2>
      <p>Fill in all fields and click "create event"</p>

      {/* form starts */}
      <form onSubmit={handleCreateEvent}>
        <div className="eventName">
          <label htmlFor="eventName">
            Event Name <br />
            <input
              type="text"
              id="eventName"
              name="eventName"
              value={fields.eventName}
              onChange={handleFieldChange}
              placeholder="My special event"
            />
          </label>
        </div>

        <div className="hostName">
          <label htmlFor="hostName">
            Host Name <br />
            <input
              type="text"
              id="hostName"
              name="hostName"
              value={fields.hostName}
              onChange={handleFieldChange}
              placeholder="Nigella Lawson"
            />
          </label>
        </div>

        <div className="date">
          <label htmlFor="date">
            Date <br />
            <input
              type="date"
              id="date"
              name="date"
              value={fields.date}
              onChange={handleFieldChange}
            />
          </label>
        </div>

        <div className="time">
          <label htmlFor="time">
            Time <br />
            <input
              type="time"
              id="time"
              name="time"
              value={fields.time}
              onChange={handleFieldChange}
            />
          </label>
        </div>

        <div className="address">
          <label htmlFor="address">
            Address <br />
            <input
              type="text"
              id="address"
              name="address"
              value={fields.address}
              onChange={handleFieldChange}
              placeholder="Mi casa"
            />
          </label>
        </div>

        <div className="createEventButton">
          <button className="submitButton" type="submit">
            Create event
          </button>
        </div>
        <Alert message={alert.message} success={alert.isSuccess} />
      </form>
      {/* form ends */}
    </div>
  );
};

export default CreateEvent;
