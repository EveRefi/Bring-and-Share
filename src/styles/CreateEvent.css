@import url("https://fonts.googleapis.com/css2?family=Roboto&display=swap");

.createEvent {
  font-family: "Roboto", sans-serif;
  display: flex;
  flex-direction: column;
  justify-content: center;
  text-align: center;
  margin: 0px;
}

.createEventTitle {
  margin: 10px;
  font-size: 25px;
  color: rgb(255, 89, 0);
}

.createEventInstruction {
  margin: 8px;
  font-size: 20px;
}

.eventName,
.hostName,
.date,
.time,
.address,
.hostEmail,
.createEventButton {
  width: 200px;
  padding: 5px;
  margin-left: auto;
  margin-right: auto;
  font-weight: 700;
  font-size: 16px;
  color: rgb(201, 105, 2);
}

#eventName,
#hostName,
#date,
#time,
#address,
#hostEmail {
  height: 22px;
  border-radius: 8px;
  border: 1px solid rgb(102, 102, 102);
}

.submitButton {
  margin-bottom: 10px;
  padding: 10px;
  border-radius: 10px;
  color: rgb(4, 4, 4);
  font-weight: 700;
  background-color: rgb(250, 151, 58);
  border: none;
  box-shadow: 5px 5px #888888;
}

/*media query start*/
@media screen and (min-width: 500px) {
  .createEventForm {
    width: 500px;
    margin-left: auto;
    margin-right: auto;
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: space-evenly;
  }

  .eventName,
  .hostName,
  .date,
  .time {
    width: 200px;
    padding: 7px;
    font-size: 18px;
  }
  .address,
  .hostEmail,
  .createEventButton {
    width: 300px;
    padding: 7px;
    font-size: 18px;
  }

  #eventName,
  #hostName,
  #date,
  #time {
    height: 24px;
    width: 170px;
  }
  #address,
  #hostEmail {
    height: 24px;
    width: 200px;
  }
}

@media screen and (min-width: 800px) {
  .createEventForm {
    width: 650px;
  }

  .eventName,
  .hostName,
  .date,
  .time {
    width: 280px;
    padding: 10px;
    font-size: 20px;
  }
  .address,
  .hostEmail,
  .createEventButton {
    width: 400px;
    padding: 10px;
    font-size: 20px;
  }

  #eventName,
  #hostName,
  #date,
  #time {
    height: 27px;
    width: 200px;
  }
  #address,
  #hostEmail {
    height: 27px;
    width: 260px;
  }
}
/*media query end*/
