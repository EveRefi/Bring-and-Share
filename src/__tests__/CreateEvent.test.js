import React from "react";
import { render } from "@testing-library/react";
import CreateEvent from "../components/CreateEvent";

describe('CreateEvent component', () => {

    it('renders correctly', () =>{

        const {asFragment} = render (<CreateEvent/>)

        expect(asFragment()).toMatchSnapshot()
    })

    it('renders the correct form', () => {
        const { getByTestId } = render (<CreateEvent/>)

        expect(getByTestId('eventName')).toBeValid()
        expect(getByTestId('hostName')).toBeValid()
        expect(getByTestId('date')).toBeValid()
        expect(getByTestId('time')).toBeValid()
        expect(getByTestId('address')).toBeValid()
        expect(getByTestId('eventName')).toBeInTheDocument()
    })
})