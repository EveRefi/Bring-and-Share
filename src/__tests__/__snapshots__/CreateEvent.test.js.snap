// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateEvent component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="createEvent"
  >
    <h2
      class="createEventTitle"
    >
      Create your event
    </h2>
    <p
      class="createEventInstruction"
    >
      Fill in all fields and click "create event"
    </p>
    <form
      class="createEventForm"
    >
      <div
        class="eventName"
      >
        <label
          for="eventName"
        >
          Event Name 
          <br />
          <input
            data-testid="eventName"
            id="eventName"
            name="eventName"
            placeholder="My special event"
            type="text"
            value=""
          />
        </label>
      </div>
      <div
        class="hostName"
      >
        <label
          for="hostName"
        >
          Host Name 
          <br />
          <input
            data-testid="hostName"
            id="hostName"
            name="hostName"
            placeholder="Nigella Lawson"
            type="text"
            value=""
          />
        </label>
      </div>
      <div
        class="date"
      >
        <label
          for="date"
        >
          Date 
          <br />
          <input
            data-testid="date"
            id="date"
            name="date"
            type="date"
            value=""
          />
        </label>
      </div>
      <div
        class="time"
      >
        <label
          for="time"
        >
          Time 
          <br />
          <input
            data-testid="time"
            id="time"
            name="time"
            type="time"
            value=""
          />
        </label>
      </div>
      <div
        class="address"
      >
        <label
          for="address"
        >
          Address 
          <br />
          <input
            data-testid="address"
            id="address"
            name="address"
            placeholder="Mi casa"
            type="text"
            value=""
          />
        </label>
      </div>
      <div
        class="hostEmail"
      >
        <label
          for="hostEmail"
        >
          Host's email 
          <br />
          <input
            data-testid="hostEmail"
            id="hostEmail"
            name="hostEmail"
            placeholder="hostEmail"
            type="text"
            value=""
          />
        </label>
      </div>
      <div
        class="createEventButton"
      >
        <button
          class="submitButton"
          data-testid="submitButton"
          type="submit"
        >
          Create event
        </button>
      </div>
      <div
        class="Alert alert-error"
      />
    </form>
  </div>
</DocumentFragment>
`;
